File: manashelper-bot/src/main/java/kg/manasuniversity/usbtypec/manashelper/client/TimetableClient.java
package kg.manasuniversity.usbtypec.manashelper.client;

import org.springframework.stereotype.Service;
import org.springframework.web.client.RestClient;

@Service
public class TimetableClient {
    private static final String TIMETABLE_BASE_URL = "http://timetable.manas.edu.kg";
    private static final String SPECIFIC_COURSE_URL = "/department-printer/{id}";

    private final RestClient restClient = RestClient.create(TIMETABLE_BASE_URL);

    public String fetchTimetableHtml(int courseId) {
        return restClient.get()
            .uri(SPECIFIC_COURSE_URL, courseId)
            .retrieve()
            .body(String.class);
    }
}

File: src/manashelper/scraping/timetable_client.py
import httpx

TIMETABLE_URL_TEMPLATE = "http://timetable.manas.edu.kg/department-printer/{course_id}"


class TimetableClient:
    def __init__(self, http_client: httpx.AsyncClient) -> None:
        self._http_client = http_client

    async def fetch_timetable_html(self, course_id: int) -> str:
        response = await self._http_client.get(TIMETABLE_URL_TEMPLATE.format(course_id=course_id))
        response.raise_for_status()
        return response.text
